packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -597,7 +597,9 @@ abstract class BucketBase extends Resource implements IBucket {
     }
 
     if (this.encryptionKey) {
-      this.encryptionKey.grant(grantee, ...keyActions);
+      if (keyActions.length !== 0 || keyActions === undefined) {
```

Comment:
This will actually error if `keyActions` is `undefined`. To you really want to pass `undefined` to `encryptionKey.grant`? I suggest just:

```typescript
if (this.encryptionKey && keyActions && keyActions.length !== 0) {
  this.encryptionKey.grant(grantee, ...keyActions);
}
```

Reply:
Sorry, my bad. I'm not sure what I was thinking there. You are absolutely right, `keyActions === undefined` check is totally wrong in this context. I have fixed my mistake.

Thanks for your review @iliapolo

Reply:
👍